### PR TITLE
[projects] enable New Workspace button in all cases

### PR DESCRIPTION
**Pull request by AlexTugarev**
Fixes https://github.com/gitpod-io/gitpod/issues/5591

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#5591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by AlexTugarev** [comment]
@gtsiolis and @svenefftinge while working on this and trying to reproduce the initial issue with an invalid config, I see that the initial config isn't marked as invalid anymore. I wonder about any changes were made in the meantime to this area. 

Screen Shot 2021-09-13 at 09 13 49

> Allow users to open a new workspace when configuration is invalid.

When should it be enabled now? For testing here I enabled it without conditions. When should it actually be disabled?

**Comment by gtsiolis** [comment]
> I see that the initial config isn't marked as invalid anymore. I wonder about any changes were made in the meantime to this area.

@AlexTugarev I could reproduce this on load! However, I was able to reproduce see the invalid config error after clicking _Run Prebuild_ for a project with invalid configuration.

> When should it actually be disabled?

A boring (simple) solution could be to keep the _New Workspace_ button always enabled as disabling buttons is also a bad accessibility practice. If we'd like to disable the _New Workspace_ button we could do it only when someone adds a new project but there's no git-based configuration.

**Comment by AlexTugarev** [comment]
@gtsiolis, please have another pass on this. The SVG is still missing ;-)

**Review by gtsiolis — APPROVED** [review]
Seems to resolve #5591, although I'd expect this flow to be improved in future iteretions ✔️ 

Thanks for tackling this, @AlexTugarev!

Approving to unblock merging but holding in case you need to make any further changes.

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 662686b1f1229b613a0b89631b68b9e14fae4fe8</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5654#pullrequestreview-756716202" title="Approved">gtsiolis</a>*

Associated issue: *#5591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5654#pullrequestreview-756716202" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5654#issuecomment-921590305" title="LGTM">JanKoehnlein</a>*

Associated issue: *#5591*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein,gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by jankeromnes** [comment]
⚠️ There are a few problems with always enabling the `New Workspace` button:

1. It only actually uses the configuration visible on the left when:
    a. You ran a prebuild, and it stopped successfully
    b. You already had a Git-based config
2. However, it might use something totally different if:
    a. You ran a prebuild, but it didn't finish yet
    b. You ran a prebuild, but it had an error
    c. You didn't run a prebuild, and there was no Git-based config

I don't think it's a good idea to enable all of 1. and 2. Maybe it's safer to enable just 1.a. (previous code) and 1.b. (new request from https://github.com/gitpod-io/gitpod/issues/5591)?

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (jankeromnes) [inline components/dashboard/src/projects/ConfigureProject.tsx:182]: Instead of always enabling `New Workspace`, I propose to only enable it when:

- There was a Git-based config (new with this PR, requested by https://github.com/gitpod-io/gitpod/issues/5591)
- User ran a Prebuild, which had no errors, and stopped successfully (previous code)

I think we don't want to enable the `New Workspace` button when:

- User ran a Prebuild, but it's not quite finished yet (so starting a workspace wouldn't have the config)
- User ran a Prebuild, but it had an error (starting a workspace would not use any prebuild)
- User didn't run a Prebuild yet, and config was auto-detected (what you see in the editor isn't saved yet, and starting a new workspace would give you an unconfigured workspace)

```suggestion
                    {((!isDetecting && isEditorDisabled) || (prebuildInstance?.status.phase === "stopped" && !prebuildInstance?.status.conditions.failed))
              ? <a className="my-auto" href={`/#${project?.cloneUrl}`}><button className="secondary">New Workspace</button></a>
              : <button disabled={true} className="secondary">New Workspace</button>}
```

Comment (gtsiolis) [reply components/dashboard/src/projects/ConfigureProject.tsx:182]: Great catch and proposal @jankeromnes!  💯 

All points are valid. Have also verified most of the issues you mentioned in https://github.com/gitpod-io/gitpod/pull/5654#issuecomment-921665768.

**Comment by AlexTugarev** [comment]
> There was a Git-based config (new with this PR, requested by #5591)

@gtsiolis  and @jankeromnes, please help to reproduce it. last time I checked, there was no validation at all.

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/ConfigureProject.tsx:182]: Now implemented in https://github.com/gitpod-io/gitpod/pull/5907

**Pull request closed without merging**
